src/chocolatey/infrastructure/services/XmlService.cs

```diff
@@ -65,30 +98,45 @@ public void serialize<XmlType>(XmlType xmlType, string xmlFilePath, bool isSilen
         {
             _fileSystem.create_directory_if_not_exists(_fileSystem.get_directory_name(xmlFilePath));
 
-            var xmlUpdateFilePath = xmlFilePath + ".update";
-
             FaultTolerance.try_catch_with_logging_exception(
                 () =>
                 {
                     var xmlSerializer = new XmlSerializer(typeof(XmlType));
-                    //var textWriter = new StreamWriter(xmlUpdateFilePath, append: false, encoding: new UTF8Encoding(encoderShouldEmitUTF8Identifier: false))
-                    var textWriter = new StreamWriter(xmlUpdateFilePath, append: false, encoding: new UTF8Encoding(encoderShouldEmitUTF8Identifier: true))
-                    {
-                        AutoFlush = true
-                    };
 
-                    xmlSerializer.Serialize(textWriter, xmlType);
-                    textWriter.Flush();
+                    // Write the updated file to memory
+                    using(var memoryStream = new MemoryStream())                        
+                    using(var textWriter = new StreamWriter(memoryStream, encoding: new UTF8Encoding(encoderShouldEmitUTF8Identifier: true)))

```

Comment:
Also instead of `using`, I actually prefer to be explicit when it comes to streams.